restore_memory.py

```
@@ -0,0 +1,195 @@
+            print(f"Removing current database: {DB_PATH}")
+            shutil.rmtree(db_path)
+
+        # Extract backup
+        print(f"Extracting backup: {backup_file}")
+        db_path.mkdir(parents=True, exist_ok=True)
+
+        with zipfile.ZipFile(backup_path, 'r') as zip_ref:
+            zip_ref.extractall(db_path)
+
+        print("Restore completed successfully!")
+        return True
```

Ruff (0.14.14)

Consider moving this statement to an `else` block

(TRY300)